gate-oauth2/src/main/java/com/netflix/spinnaker/gate/security/oauth2/SpinnakerUserInfoTokenServices.java

```diff
@@ -0,0 +1,304 @@
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.security.oauth2.resource.ResourceServerProperties;
+import org.springframework.boot.autoconfigure.security.oauth2.resource.UserInfoTokenServices;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.security.oauth2.common.exceptions.InvalidTokenException;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.OAuth2Request;
+import org.springframework.security.oauth2.provider.token.ResourceServerTokenServices;
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken;
+
+/**
+ * ResourceServerTokenServices is an interface used to manage access tokens. The
+ * UserInfoTokenService object is an implementation of that interface that uses an access token to
+ * get the logged in user's data (such as email or profile). We want to customize the Authentication
+ * object that is returned to include our custom (Kork) User.
+ */
+@Slf4j
+public class SpinnakerUserInfoTokenServices implements ResourceServerTokenServices {
+
+  @Autowired private ResourceServerProperties sso;
+  @Autowired private UserInfoTokenServices userInfoTokenServices;
+  @Autowired private CredentialsService credentialsService;
+  @Autowired protected OAuth2SsoConfig.UserInfoMapping userInfoMapping;
+  @Autowired protected OAuth2SsoConfig.UserInfoRequirements userInfoRequirements;
+  @Autowired private PermissionService permissionService;
+  @Autowired private Front50Service front50Service;
+
+  @Autowired(required = false)
+  private SpinnakerProviderTokenServices providerTokenServices;
+
+  @Autowired private AllowedAccountsSupport allowedAccountsSupport;
+  @Autowired private FiatClientConfigurationProperties fiatClientConfigurationProperties;
+  @Autowired private Registry registry;
+
+  @Autowired(required = false)
+  @Qualifier("spinnaker-oauth2-group-extractor")
+  private BiFunction<String, Map, List<String>> groupExtractor;
+
+  private RetrySupport retrySupport = new RetrySupport();
+
+  @Override
+  public OAuth2Authentication loadAuthentication(final String accessToken)
+      throws AuthenticationException, InvalidTokenException {
+    OAuth2Authentication oAuth2Authentication =
+        userInfoTokenServices.loadAuthentication(accessToken);
+
+    final Map<String, Object> details =
+        (Map<String, Object>) oAuth2Authentication.getUserAuthentication().getDetails();
+
+    log.debug("UserInfo details: " + details);
```

Comment:
The groovy code has:
```
    if (log.isDebugEnabled()) {
      log.debug("UserInfo details: " + entries(details))
    }
```
What does the `entries` method do?  What do the log messages look like before/after?

Reply:
https://github.com/spinnaker/gate/pull/1871/commits/304180e0876a85b97006fbf5e82370cfeb6ddd28

Reply:
Great.  Can you put back the
```
if (log.isDebugEnabled()) {
```
too please?

Reply:
https://github.com/spinnaker/gate/pull/1871/commits/db0949aa085a34d209963ac6e2176fcee362e2b7